osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -124,16 +124,19 @@ public void UpdateBeatmap(BeatmapInfo beatmap)
             // todo: this method should be smarter as to not recreate panels that haven't changed, etc.
             var group = groups.Find(b => b.BeatmapSet.ID == set.ID);
 
-            if (group == null)
-                return;
-
             int i = groups.IndexOf(group);
-            groups.RemoveAt(i);
+            if (i >= 0)

```

Comment:
how does this happen?

Reply:
If we hide all beatmaps in a set it can't find the group anymore. So the idea was to remove it when it's still there and readd it in the same spot once a new group was successfully created or just add it if there is no index available.

Reply:
right, so this fixes restoring completely hidden groups without exiting-reentering song select?

Reply:
Yes. This causes them to reappear normally.